src/partition/sph-partition-matches.cc

```diff
@@ -160,54 +217,124 @@ Worker::Worker(ConfigStore const& config)
 
 void Worker::map(char const* const begin, char const* const end, Worker::Silo& silo) {
     std::pair<double, double> sc1, sc2;
-    ChunkLocation loc1, loc2;
     char const* cur = begin;
     while (cur < end) {
         cur = _editor.readRecord(cur, end);
-        bool null1 = _editor.isNull(_pos1.first) || _editor.isNull(_pos1.second);
-        bool null2 = _editor.isNull(_pos2.first) || _editor.isNull(_pos2.second);
-        if (null1 && null2) {
-            throw std::runtime_error(
-                    "Both partitioning positions in a match "
-                    "record contain NULLs.");
-        }
-        if (!null1) {
-            sc1.first = _editor.get<double>(_pos1.first);
-            sc1.second = _editor.get<double>(_pos1.second);
-            loc1 = _chunker.locate(sc1);
-        }
-        if (!null2) {
-            sc2.first = _editor.get<double>(_pos2.first);
-            sc2.second = _editor.get<double>(_pos2.second);
-            loc2 = _chunker.locate(sc2);
-        }
-        if (!null1) {
-            _editor.set(_chunkIdField, loc1.chunkId);
-            _editor.set(_subChunkIdField, loc1.subChunkId);
+        if (_id1Field != -1) {
```

Comment:
This is a very long if/else with some code duplication and non-trivial control flow. It's hard to follow and verify that there is no mixup in variable names (sc1/sc2, loc1/loc2, etc). Some of these code could probably be moved to a separate method/class.